celery/backends/dynamodb.py

```diff
@@ -206,6 +225,156 @@ def _get_or_create_table(self):
             else:
                 raise e
 
+    def _has_ttl(self):
+        """Return the desired Time to Live config; True means enabled."""
+        return self.time_to_live_seconds is not None
+
+    def _get_ttl_specification(self, ttl_attr_name):
+        """Get the boto3 structure describing the DynamoDB TTL specification."""
+        return {
+            'TableName': self.table_name,
+            'TimeToLiveSpecification': {
+                'Enabled': self._has_ttl(),
+                'AttributeName': ttl_attr_name
+            }
+        }
+
+    def _set_table_ttl(self):
+        """Enable or disable Time to Live on the table."""
+
+        # Verify that the client supports the TTL methods.
+        for method in ('update_time_to_live', 'describe_time_to_live'):
+            if not hasattr(self._client, method):
+                message = (
+                    "boto3 method '{method}' not found; ensure that "
+                    "boto3>=1.9.178 and botocore>=1.12.178 are installed"
+                ).format(method=method)
+                if not self._has_ttl():
+                    # Return if Time to Live is not desired anyway.
+                    logger.debug(message)
+                    return
+                else:
+                    # Raise exception if Time to Live should be enabled.
+                    logger.error(message)
+                    raise AttributeError(
+                        "boto3 method '{}' not found".format(method)
+                    )
+
+        # Get the current TTL description.
+        try:
+            description = self._client.describe_time_to_live(
+                TableName=self.table_name
+            )
+            status = description['TimeToLiveDescription']['TimeToLiveStatus']
+        except ClientError as e:
+            error_code = e.response['Error'].get('Code', 'Unknown')
+            error_message = e.response['Error'].get('Message', 'Unknown')
+            logger.error((
+                'Error describing Time to Live on DynamoDB table {table}: '
+                '{code}: {message}'
+            ).format(
+                table=self.table_name,
+                code=error_code,
+                message=error_message,
+            ))
+            raise e
```

Comment:
This method has a considerable size, so perhaps you could extract these lines to a different method (no need to modify tests):

```python     
def _get_table_ttl_description(self):
    ...
    return description
```

And its use:

```python
description = self._get_table_ttl_description()
status = description['TimeToLiveDescription']['TimeToLiveStatus']
```

How does that sound?

Reply:
Addressed.